src/MergeIntervals.js

```diff
@@ -0,0 +1,22 @@
+const merge = function (intervals) {
+  if (intervals.length === 1) return intervals;
+
+  intervals.sort((a, b) => a[0] - b[0]);
+
+  const result = [intervals[0]];
+  let currentIndex = 0;
+
+  for (let i = 1; i < intervals.length; i++) {
+    if (result[result.length - 1][1] >= intervals[i][0]) {
+      result[result.length - 1] = [
+        result[result.length - 1][0],
+        Math.max(result[result.length - 1][1], intervals[i][1]),
+      ];
```

Comment:
```suggestion
      result[result.length - 1][1] = Math.max(result[result.length - 1][1], intervals[i][1]);      
```

`result[result.length - 1][0]`이 바뀌는 경우의 수가 없는거 같아서 이렇게 하는게 나을거 같습니다.